app/controllers/api/base_controller/providers.rb

```diff
@@ -46,6 +46,26 @@ def delete_resource_providers(type, id = nil, _data = nil)
         end
       end
 
+      def meta_data_resource_providers(_type, _id = nil, data)
+        provider = ExtManagementSystem.find_by_name(data.delete("name"))
+        data.each do |key, value|
+          attribute = provider.custom_attributes.find_by_name(key)
+          if !attribute.blank?
+            attribute.update(value)
+          elsif value["field_type"]
+            provider.custom_attributes << CustomAttribute.create(:name       => key,
+                                                                 :value      => value["value"],
+                                                                 :field_type => value["field_type"])
+          else
+            provider.custom_attributes << CustomAttribute.create(:name  => key,
+                                                                 :value => value["value"])
+          end
+        end
+        provider.custom_attributes
+      rescue => err
+        raise BadRequestError, "Could not update the provider custom attributes - #{err}"
```

Comment:
Not sure what can blow up here as you don't call `update!` or `create!`...On that note, does it make sense to call the `!` versions of the methods.